core/modules/layout/layout.flexible.inc

```diff
@@ -0,0 +1,1030 @@
+  // is always selected) but seems we cannot set #default_value for a field if
+  // its in $form_state['input']. https://drupal.stackexchange.com/questions/251056/set-default-form-values-after-ajax-callback
+  $default_value = key($options);
+  unset($form_state['input']['region_style']);
+
+  $form['row_settings']['region_style'] = array(
+    '#title' => t('Region widths'),
+    '#type' => 'radios',
+    '#default_value' => key($options),
+    '#options' => $options,
+  );
+
+  $form['submit_region_style'] = array(
+    '#type' => 'submit',
+    '#value' => t('Choose region widths'),
+    '#attributes' => array('class' => array('layout-title-button')),
+    '#attributes' => array(
+      'class' => array('js-hide'),
+    ),
+    '#submit' => array(
+      'layout_flexible_template_region_count_load',
+    ),
+  );
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#name' => 'submit',
+    '#value' => t('Continue'),
+    '#attributes' => array('class' => array('layout-title-button')),
+    '#submit' => array(
+      'layout_flexible_template_edit_row_submit',
+    ),
+    '#ajax' => array(
+      'callback' => 'layout_ajax_form_open_dialog',
+    ),
+  );
+
+  return $form;
+}
+
+/**
+ * Submit handler for layout_flexible_template_region_style_select().
+ */
+function layout_flexible_template_edit_row_submit($form, &$form_state) {
+  $form_state['redirect'] = 'admin/structure/layouts/settings/flexible-template/' . $form_state['flexible_template_id'] . '/row/' . $form_state['original_row'] . '/configure/' . $form_state['values']['region_style'];
+}
+
+/**
+ * Form to add a row to a flexible template.
+ *
+ * @param $flexible_template_id
+ *   The template to which a row is being added.
+ * @param $original_row
+ *   The row above or below which a new row is being inserted.
+ * @param $op
+ *   The operation being performed on this row.
+ * @param $region_style
+ *   The selected region style.
+ *
+ * @ingroup forms
+ */
+function layout_flexible_template_edit_row_form($form, &$form_state, $flexible_template_id, $original_row, $region_style = NULL) {
+  form_load_include($form_state, 'inc', 'layout', 'layout.admin');
+
+  $form_state['flexible_template_id'] = $flexible_template_id;
+  $form_state['original_row'] = $original_row;
+  $form_state['template_data'] = layout_flexible_template_template($flexible_template_id);
+
+  $row_data = array();
+  if ($original_row != 'add') {
+    $row_data = $form_state['template_data']['regions'][$original_row];
+  }
+
+  $region_style = $region_style ? $region_style : $row_data['contains'];
+  $form_state['region_style'] = $region_style;
+
+  $row_styles = layout_flexible_row_styles();
+  $selected_style = $row_styles[$region_style]['name'];
+  $region_count = $row_styles[$region_style]['region_count'];
+
+  if ($original_row == 'add') {
+    backdrop_set_title(t('Add new row'));
+  }
+  else {
```

Comment:
This can go before the if statement without the else.

Reply:
I would prefer an if/else if each block called a function. We'd be calling `backdrop_set_title()` twice.

Reply:
But you don't need to call it twice. Create a `$title = ...` and then use that in `backdrop_set_title()`.

Reply:
Yes but we dont gain anything from that, we still require a conditional and introduce an unnecessary variable.